scripts/enable-ecs-agent-gpu-support.sh

```diff
@@ -1,6 +1,47 @@
 #!/usr/bin/env bash
 set -ex
 
+if [[ $AMI_TYPE != "al2gpu" ]]; then
+    exit 0
+fi
+
+GPG_CHECK=1
+# don't do the gpg check in air-gapped regions
+if [ -n "$AIR_GAPPED" ]; then
+    GPG_CHECK=0
+fi
+tmpfile=$(mktemp)
+cat >$tmpfile <<EOF
+[amzn2-nvidia]
+name=Amazon Linux 2 Nvidia repository
+mirrorlist=\$awsproto://\$amazonlinux.\$awsregion.\$awsdomain/\$releasever/amzn2-nvidia/latest/\$basearch/mirror.list
+priority=20
+gpgcheck=$GPG_CHECK
+gpgkey=https://developer.download.nvidia.com/compute/cuda/repos/rhel7/x86_64/7fa2af80.pub
+enabled=1
+exclude=libglvnd-*
+EOF
+
+# this repo is temporary and only used for installing the system-release-nvidia package
+sudo mv $tmpfile /etc/yum.repos.d/amzn2-nvidia-tmp.repo
+# system-release-nvidia creates an nvidia repo file at /etc/yum.repos.d/amzn2-nvidia.repo
+sudo yum install -y system-release-nvidia
+sudo rm /etc/yum.repos.d/amzn2-nvidia-tmp.repo
+
+sudo yum install -y kernel-devel-$(uname -r) \
+    system-release-nvidia \
+    nvidia-driver-latest-dkms \
+    kmod-nvidia-latest-dkms \
+    pciutils \
+    xorg-x11-server-Xorg \
+    docker-runtime-nvidia \
+    oci-add-hooks \
+    libnvidia-container \
+    libnvidia-container-tools \
+    nvidia-container-runtime-hook \
+    cuda-drivers-fabricmanager \
```

Comment:
Q. would these packages be installed in any order? and we won't encounter the following issue mentioned in the current AL2 graphic recipe? 
```
{# install cuda after other packages are installed #}
{# otherwise the install fails due to yum not realizing #}
```
Thanks!

Reply:
When installing multiple packages with a single command, yum will evaluate all package dependencies and install in the correct order.

So if a package was not stating it's dependencies correctly that might cause problems, but I'm not aware of any of these packages having incorrect dependencies at the moment, might be that a previous issue with the cuda package has been fixed now.